packages/medusa-cli/src/commands/link.js

```diff
@@ -48,29 +52,70 @@ module.exports = {
         process.exit(1)
       })
 
+    const linkActivity = logger.activity("Linking local project")
+
     // Create the user with the user id
     if (!argv.skipLocalUser && auth.user) {
-      const localCmd = resolveLocalCommand(`user`)
-      await localCmd({
-        directory: argv.directory,
-        id: auth.user.id,
-        email: auth.user.email,
-      })
+      let proc
+      try {
+        proc = await execa(
+          `medusa`,
+          [`user`, `--id`, auth.user.id, `--email`, auth.user.email],
+          {
+            env: {
+              ...process.env,
+              NODE_ENV: "command",
+            },
+          }
+        )
+      } catch (error) {
+        logger.failure(linkActivity, "Failed to perform local linking")
+        process.exit(1)
+      }
     }
 
-    // This step sets the Cloud link by opening a browser
-    const bo = await open(
-      `${appHost}/local-link?lurl=http://localhost:9000&ltoken=${auth.user.id}`,
+    logger.success(linkActivity, "Local project linked")
+
+    console.log()
+    console.log(
+      "Link Medusa Cloud to your local server. This will open the browser"
+    )
+    console.log()
+
+    const qs = [
       {
+        type: "input",
+        name: "open",
+        message: "Press enter key to open browser for linking or n to exit",
+      },
+    ]
+
+    await inquirer.prompt(qs).then(async a => {
+      if (a.open === "n") {
+        process.exit(0)
+      }
+
+      const params = `lurl=http://localhost:${port}&ltoken=${auth.user.id}`
+
+      // This step sets the Cloud link by opening a browser
+      const bo = await open(`${appHost}/local-link?${encodeURI(params)}`, {
```

Comment:
If I can add a comment: It would be nice if we could practice more elaborate naming. It can be hard for people new to these kinds of files or people who haven't been introduced to instantly understand what happens.

what does `bo` stand for? The same goes for `qs` above :D Much appreciated